tsdb/engine/tsm1/engine.go

```diff
@@ -1093,9 +1101,15 @@ func (e *Engine) CreateSnapshot() (string, error) {
 	}
 
 	e.mu.RLock()
-	defer e.mu.RUnlock()
+	path, err := e.FileStore.CreateSnapshot()
+	if err != nil {
+		e.mu.RUnlock()
+		return "", err
+	}
+	e.mu.RUnlock()
```

Comment:
Looks like this was a pre-existing bug in that case. I'll add the lock protection.